std/string.d

```diff
@@ -221,10 +221,66 @@ inout(char)[] fromStringz(inout(char)* cString) @nogc @system pure nothrow {
 ///
 @system pure unittest
 {
-    assert(fromStringz(null) == null);
```

Comment:
I've also run into the problem of `null` literals matching multiple overrides. Is there a way to get the D compiler to resolve plain `null` to a specific override when there are multiple potentially matching functions without writing a `cast` every time?

Reply:
Create an override that accepts void?

Reply:
For this particular case I've found what looks like a solution, changing `toStringz` to a templated function with a default template argument:

https://run.dlang.io/is/yJSJhX
```d
import std.stdio;

inout(Char)[] fromStringz(Char = char)(inout(Char)* cString)
if (is(Char == char) || is(Char == wchar) || is (Char == dchar))
{
	import core.stdc.wchar_ : wchar_t;
    static if (Char.sizeof == 1)
        import core.stdc.string : cstrlen = strlen;
    else static if (Char.sizeof == wchar_t.sizeof)
        import core.stdc.wchar_ : cstrlen = wcslen;
    else
        static size_t cstrlen(const wchar* s)
    	{
        	const(wchar)* p = s;
       		while (*p)
                ++p;
        	return p - s;
        }
    return cString ? cString[0 .. cstrlen(cString)] : null;
}

void main()
{
    string s = "abc\0"c;
    wstring w = "abc\0"w;
    assert(fromStringz(s.ptr) == "abc"c);
    assert(fromStringz(w.ptr) == "abc"w);
    assert(fromStringz(null) is null); // Can use plain `null`!
    writeln("Tests passed.");
}
```

Reply:
That's a nice solution, it also implements dchar too. @wilzbach Any concerns about modifying the existing code as opposed to adding a new override?

Reply:
You should be able to use `typeof(null)` if you want a function to explicitly accept `null`, though if a default template argument solves the problem, that seems cleaner than having to add an overload just for `null`.

Reply:
@jmdavis Thanks, that may solve the problem more generally for me.